### DISCUSSION TITLE
PI controller for fans speed control?

### DISCUSSION THREAD

**Discussion by NoxaNM15R5** [T-Troll/alienfx-tools#353] [Ideas]
Hi!
Thank you for this lovely program which gives freedom to control fans and keyboard light. This is marvelous!
I have a peculiar case for alienfan application. My M16 has a powerful 13900x CPU that gets hot when concentrated work is loaded. The temperatures of the CPU jump from 65 to 75 intermittently and go back to low levels when the sudden job is done.

When using a linear curve in alienfan, these jumping temperatures reflect as quickly changing fan speed hence generating acoustics of different frequencies which disturbs me after a while. Imagine fans going slow -> fast -> slow in a few seconds.
Therefore I was thinking of a system like an integrated controller which checks past temperatures to spin up the system accordingly. The system will respond slowly to intermittent changes but will respond to past averages.
Also, this could be done by using the average temperature of multiple sensors instead of monitoring one fast-switching sensor. This will bring a problem of neglecting max temps but if we set a low prochot value, I think frying the CPU could be avoided.

**Comment by T-Troll** [comment]
Good idea, but don't forget about temp spikes. So i try to keep fans AS FAST as possible, it's quite inertial (and it's vital for the most systems having 100C peak).
But i think about other modes as well.

**Comment by Torduu** [comment]
Do you know about undervolting? MashIT on youtube has a really nice guide that helped alot with my temps. 
In case it can further help you. Have a good day

**Comment by T-Troll** [comment]
Yes, i know and use it. ThrottleStop works well, so i don't want to reproduce it's functionality (except turbo autotune).
